### ISSUE TITLE
[Linux] Tray menu not working

### ISSUE THREAD

**Issue by anatoli26** [keepassxreboot/keepassxc#102]
When the app is minimized to tray, the tray menu on Ubuntu 16.04 is not working, i.e. clicking on the icon shows the menu, but the "Toggle window" and "Quit" options have no effect on the app. The window doesn't appear, nor the app exits on Quit.

At the same time, the auto-type combination works well, i.e. it brings to front the credentials window.

**Comment by TheZ3ro** [comment]
I think you are on the same macchine and you use Unity as DesktopEnvironment but please fill this form

```
## Your Environment
* KeePassXR version/commit used: 
* Qt version (e.g. Qt 5.3): 
* Compiler (e.g. Clang++3.6.0): 
* Operating System and version: 
* Desktop Environment:
```

I think it's a problem under unity or your desktop environment. On my XFCE it works fine.
When I know your DE I can do some test to reproduce this bug

**Comment by anatoli26** [comment]
@TheZ3ro, here it goes:
* KeePassXR version/commit used: 29f9359
* Qt version (e.g. Qt 5.3): Qt version 5.5.1, QMake version 3.0
* Compiler (e.g. Clang++3.6.0): gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4)
* Operating System and version: Ubuntu 16.04.1 LTS
* Desktop Environment: Unity 7.4.0
* Tray: libappindicator1 (12.10.1+15.04.20141110-0ubuntu1) / libappindicator3-1 (12.10.1+15.04.20141110-0ubuntu1)

**Comment by TheZ3ro** [comment]
This seems related to https://github.com/keepassxreboot/keepassxc/issues/73

Please @anatoli26 check that Issue

**Comment by anatoli26** [comment]
Indeed, related. Please see an update in [my comment](https://github.com/keepassxreboot/keepassxc/issues/73#issuecomment-263072231)  for more details.

**Comment by pseudolog** [comment]
Just wanted to confirm this behaviour.

This [bugreport](https://bugs.launchpad.net/appmenu-qt5/+bug/1574699) indicates a problem with `QSystemTrayIcon::isSystemTrayAvailable();`

Commenting `QSystemTrayIcon::isSystemTrayAvailable();` in [`gui/MainWindow.cpp`](https://github.com/keepassxreboot/keepassxc/blob/a8d9e8ba7cb34ca5911bd1fe85a30f3733dedaa7/src/gui/MainWindow.cpp#L787) does indeed fix this issue for me. I know next to nothing about QT programming, so I'm not sure which other implications this might bring.

#73

**Comment by TheZ3ro** [comment]
So it's confirmed it's an Appmenu-qt5 bug.
We can't really comment `QSystemTrayIcon::isSystemTrayAvailable()`, on other platforms this works as expected.
You need to wait for a fix by them.

**Comment by phoerious** [comment]
Can this be closed?

**Comment by anatoli26** [comment]
I guess yes, with the resolution == "won't fix", as the problem is not solved, but it appears that the root cause is in Appmenu-qt5. I'm not sure if the Appmenu-qt5 team has any plans for fixing it however, so if there's a simple workaround for KPXR maybe it should be implemented.

**Comment by anatoli26** [comment]
Just talked with one of the appmenu-qt5 maintainers (https://bugs.launchpad.net/appmenu-qt5/+bug/1574699) and this issue is solved in the repository and in Ubuntu 16.10, he'll evaluate now if to release a new version for inclusion in 16.04.

**Comment by anatoli26** [comment]
The fix for appmenu-qt5 is in xenial-proposed now, if anyone is using Ubuntu 16.04, you can test it following the instructions in the comment 7 to the original bug: https://bugs.launchpad.net/appmenu-qt5/+bug/1574699/comments/7.

**Comment by anatoli26** [comment]
The fix is in all versions of Ubuntu now (16.04, 16.10), so this particular issue is completely fixed for everyone.
